### PR TITLE
Rhmap 11065 handle mongo restarts

### PR DESCRIPTION
**Pull request by PhilipGough**
### Motivation
Events were not being correctly emitted when there was an error with the connection to Mongo. Also, because some components dependent on fh-db need to listen for the close event, which was not being emitted, this was added

### CODE REVIEW

File: lib/fhmongodb.js [hunk 1]
```diff
@@ -153,13 +153,19 @@ Database.prototype.tearUp = function (auth) {
...
 
           self.db.on('error', function (err) {
             console.warn("mongodb emits error: " + err);
             self.emit('error', err);
           });
+
+          self.db.on('close', function() {
+            var message = 'Mongo connection closed';
+            console.warn(message);
+            self.emit('close', message)
```
Comment (fh-qube) [inline lib/fhmongodb.js]: BLOCKER Add a semicolon at the end of this statement. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ASemicolon)

**Comment by fh-qube** [comment]
SonarQube analysis reported 2 issues:
* BLOCKER 1 blocker
* MINOR 1 minor

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MINOR [Reformat the code to have only one statement per line.](https://github.com/feedhenry/fh-db/blob/064907b902b77d4fd4f5a8defdff9f28e5dad9c3/lib/fhmongodb.js#L251) [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AOneStatementPerLine)

**Comment by fh-automatron** [comment]
### Pull Request Quality Guidelines

As we continue to focus on quality, here are a few things to keep in mind:

* The code changes conform to the contributing & code style guides for this component (if they exist)
* A component owner should review your change (we've tagged these above)
* Automated testing has been added or updated for this change (e.g. UART, Unit/Acceptance tests)
* If updating server component this PR should be deployed to a grid somewhere (via `fhcap-cli`) - please provide a link to this
* Ensure JIRA is up to date before & after merge (e.g. deployment notes, tester notes, release notes, fix version)
* Any SonarQube issues should be addressed where applicable (use your own judgement)

Please ping a component owner (see `fh contrib`) when your builds are complete, changes are deployed and you're ready for a review.
Fhcap PR will be generated automatically.
To deploy your changes to a cluster place comment in this PR with pattern [x][y] where x=deploy and y=clustername you wish to deploy. 
You can also deploy your change manually using fhcap-cli

**Comment by fh-qube** [comment]
SonarQube analysis reported 2 issues:
* BLOCKER 1 blocker
* MINOR 1 minor

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MINOR [Reformat the code to have only one statement per line.](https://github.com/feedhenry/fh-db/blob/e61fb23ad905a632ca84446da30e573d9e92c620/lib/fhmongodb.js#L251) [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AOneStatementPerLine)

**Comment by fh-qube** [comment]
SonarQube analysis reported 1 issue:
* MINOR 1 minor

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MINOR [Reformat the code to have only one statement per line.](https://github.com/feedhenry/fh-db/blob/e5a70c18c956c0acca2306517df7436301968539/lib/fhmongodb.js#L251) [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AOneStatementPerLine)

File: lib/fhmongodb.js [hunk 2]
```diff
@@ -153,13 +153,19 @@ Database.prototype.tearUp = function (auth) {
           if (interval) {
             clearInterval(interval);
           }
-          self.db = db.db(self.name);
```
Comment (philbrookes) [inline lib/fhmongodb.js]: This is interesting, I think I added this as some authentication scenarios were failing without it, all the unit tests seem to be passing now though. Have you tried running the testing-cloud-app with this updated fh-db?

Comment (PhilipGough) [inline lib/fhmongodb.js]: @philbrookes  No I haven't but will now. If it doesn't work its no problem, I will emit the events on the initial db object and then proceed as before. Will let you know if it passes

Comment (philbrookes) [inline lib/fhmongodb.js]: discussed with @PhilipGough over IRC, this scenario had been covered by tests he ran locally.

**Comment by philbrookes** [comment]
:+1: Reviewed.

**Comment by fh-qube** [comment]
SonarQube analysis reported 1 issue:
* MINOR 1 minor

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MINOR [Reformat the code to have only one statement per line.](https://github.com/feedhenry/fh-db/blob/c9d8c8df52ffa37ab9e0a5e313858395d98a0ba1/lib/fhmongodb.js#L252) [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AOneStatementPerLine)

**Comment by fh-qube** [comment]
SonarQube analysis reported 1 issue:
* MINOR 1 minor

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MINOR [Reformat the code to have only one statement per line.](https://github.com/feedhenry/fh-db/blob/8969f73ecc96f4a3a0af722fc6a9dfc59c03e463/lib/fhmongodb.js#L252) [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AOneStatementPerLine)

**Comment by philbrookes** [comment]
:+1: Nice, safer approach

**Comment by PhilipGough** [comment]
@philbrookes Yep, I have tested and the events are still emitted from the parent connection so no point in risking any uncaught breaking changes. Going to merge and publish. Thanks

**Pull request merged**